qpc/insights/upload.py

```diff
@@ -30,13 +30,35 @@
                                 format_upload_success)
 from qpc.request import GET, request
 from qpc.translation import _
-from qpc.utils import (validate_write_file,
+from qpc.utils import (extract_json_from_tar,
+                       validate_write_file,
                        write_file)
 
 # pylint:disable=no-member
 from requests import codes
 
 
+def verify_report_fingerprints(fingerprints, report_id):
+    """Verify that report fingerprints contain canonical facts."""
+    canonical_facts = ['insights_client_id', 'bios_uuid', 'ip_addresses',
+                       'mac_addresses', 'vm_uuid', 'etc_machine_id',
+                       'subscription_manager_id']
+    verified_fingerprints = []
+    for fingerprint in fingerprints:
+        found_facts = False
+        for fact in canonical_facts:
+            if fingerprint.get(fact):
+                found_facts = True
+                break
+        if found_facts:
+            verified_fingerprints.append(fingerprint)
+        else:
+            fingerprint.pop('metadata', None)
+            print(_(messages.INSIGHTS_REPORT_INVALID_FP %
```

Comment:
Let's not log the entire fingerprint (opposite of yupana).  Here the user is close to the data.  The fingerprint's name has more meaning to the end user.  Additionally, original source names mean something.  Let's create a log entry that is one line per fingerprint with meaningful information from meta-data.